source/BannerlordCoopInstaller/Installer.cs

```diff
@@ -0,0 +1,264 @@
+            this.InstallProgress.Value = 0;
+            foreach (DictionaryEntry resource in set)
+            {
+                if (resourceDict.ContainsKey(resource.Key.ToString()))
+                {
+                    ResourceFile resourceFile = resourceDict[resource.Key.ToString()];
+                    string fileName = $"{resource.Key}.{resourceFile.FileExtension}";
+                    this.Status.Text = "Copying " + fileName + "...";
+                    if (resourceFile.FileExtension.Equals("exe"))
+                    {
+                        File.WriteAllBytes($@"{this.BannerlordPath.Text}\Modules\Coop{resourceFile.InstallPath}\{fileName}", (byte[])resource.Value);
+                    }
+                    else if (resourceFile.FileExtension.Equals("ico"))
+                    {
+                        
+                        Icon icon = (Icon)resource.Value;
+                        using (FileStream fs = new FileStream($@"{this.BannerlordPath.Text}\Modules\Coop{resourceFile.InstallPath}\{fileName}", FileMode.Create))
+                            icon.Save(fs);
+                    }
+                    else
+                    {
+                        File.WriteAllText($@"{this.BannerlordPath.Text}\Modules\Coop{resourceFile.InstallPath}\{fileName}", resource.Value.ToString());
+                    }
+                    
+                }
+                else
+                {
+                    string fileName = resource.Key + ".dll";
+                    this.Status.Text = "Copying " + fileName + "...";
+                    File.WriteAllBytes($@"{this.BannerlordPath.Text}\Modules\Coop\bin\Win64_Shipping_Client\{fileName}" , (byte[])resource.Value);
+                }
+
+            }
+        }
+
+        /// <summary>
+        /// Disable controls, check if a version of the mod is installed; ask the user to delete it. Install the mod and change install button to finish
+        /// </summary>
+        /// <param name="sender"></param>
+        /// <param name="e"></param>
+        private void Install_Click(object sender, EventArgs e)
+        {
+            this.BannerlordPath.Enabled = false;
+            this.ShortcutCheckbox.Enabled = false;
+            this.PathBrowse.Enabled = false;
+            
+            // version exists, ask to delete it
+            if (Directory.Exists($@"{this.BannerlordPath.Text}\Modules\Coop"))
+            {
+                DialogResult dialogResult = MessageBox.Show("An older version exists and must be removed. Are you sure?", "Older Version Detected", MessageBoxButtons.YesNo);
+                if (dialogResult == DialogResult.Yes)
+                {
+                    try
+                    {
+                        //recursively delete the mod
+                        Directory.Delete($@"{this.BannerlordPath.Text}\Modules\Coop", true);
+                    }catch(Exception)
```

Comment:
```suggestion
                    }
                    catch(Exception)
```

Reply:
Also should we log/display the error? I'd be curious to see the error message as a user

Reply:
The status of the installer changes in the bottom left. The user is notified to close files and processes since the only exception that should occur when deleting files as privileged is IO exception. I'll change it to that.